SimulatedWorld_Function.R

```diff
@@ -21,13 +21,30 @@ SimulateWorld <- function(){
     ex <- extent(0.5,20.5,0.5,20.5)
     extent(temp_plain) <- ex
     xy <- coordinates(temp_plain)
-    temp_plain[] <- xy[,2]
+    #temp_plain[] <- xy[,2] # not used
+    # EW: if this is a proxy for bottom temp is 4 deg inc over 100 years too high? 
+    # EW: also these values created a surface with a 4 deg change within a year -- 
+    # -- basically equaivalent to 400-600m depth in the survey
     min <- 0.0404*y + 1.9596 #temp increases by 4 degrees over 100 years
     max <- 0.0404*y + 5.9596
-    vals <- seq(min,max,0.01)
-    vals <- vals[1:400]
-    temp_plain[] <- vals
-    temp <- raster::calc (temp_plain, fun = function(x) jitter(x,amount=1))
+    # EW: next lines commented out, replaced below
+    #temp_plain[] <- seq(min,max,length.out=400) # assign values to RasterLayer
+    #temp <- raster::calc (temp_plain, fun = function(x) jitter(x,amount=.001))
+    
+    # EW: simulate from matern spatial field. We could extend this by
+    # (1) letting matern parameters vary over time, (2) letting latitude 
+    # gradient vary over time, and (3) making the field simulation be non-independent
+    # across years. Right now this generates an independent field by year
```

Comment:
Really like this! Is this nullified if we use the ROMS environmental data? Perhaps it would be a good discussion point in the workshop.

Reply:
I think we need something like this, even when the ROMS data is brought in. Similar to what Eli (@eeholmes) had said, temperature or environmental variables won't explain 100% of the variation in distribution, and we need to have other drivers (habitat, pop dynamics) -- which would sort of be encapsulated in this kind of simulation. An IBM / agent based model would be another approach, but a lot more decisions to make